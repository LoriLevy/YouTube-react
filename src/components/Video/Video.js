/* file: src/components/Video/Video.js */
import React from 'react';
import './Video.scss';

const BASE_EMBED_URL = 'https://www.youtube.com/embed/';

 function Video(props) {
    if(!props.id) {
        return null;
    }
    
    const embedUrl = `${BASE_EMBED_URL}${props.id}?autoplay=0`;
    return(
        <div className='video-container'>
            <div className='video'>
                <iframe width={'560'} height={'350'} src={embedUrl} frameBorder='0'
                allow='autoplay; encrypted-media' allowFullScreen title={'video'}/> 
            </div>
        </div>                
    );
 }
 export default Video;