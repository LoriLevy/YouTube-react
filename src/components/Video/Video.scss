.video-container { /* parent */
    position: relative;
    width: 100%;
    height: 355px;
    &:before {
      content: "";                  /* empty string because we want to use it to create the correct spacing */
      display: block;               /* By default psuedo elements are inline, and inline elements ignore maring an padding, yeah? */
      padding-top: 56.25%;          /*  9 / 16 = 0.5625 = 56.25%. So it gives us the desired 16:9 ratio */
    }
  
    .video {
      position: absolute;   /* Child */
      top: 0;
      left: 0;
      bottom: 0;
      right: 0;
  
      .video-player {
        min-height: 480px;
        width: 100%;
        height: 100%;
      }
    }
  }