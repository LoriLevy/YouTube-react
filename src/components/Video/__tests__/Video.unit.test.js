import React from 'react';
import Video from '../Video';
import {shallow} from 'enzyme';

describe('Video', () => {
    test('renders video component correctly', () => {
      const wrapper = shallow(
        <Video id='DqUQW3xyQ1c'/>
      );
      expect(wrapper).toMatchSnapshot();
    });

    test('renders null if no id in video component', ()=>{
      const wrapper = shallow( <Video/>);
      expect(wrapper).toMatchSnapshot();
    });
});
