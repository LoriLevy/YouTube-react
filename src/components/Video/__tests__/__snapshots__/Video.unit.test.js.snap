// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video renders null if no id in video component 1`] = `""`;

exports[`Video renders video component correctly 1`] = `
<div
  className="video-container"
>
  <div
    className="video"
  >
    <iframe
      allow="autoplay; encrypted-media"
      allowFullScreen={true}
      frameBorder="0"
      height="350"
      src="https://www.youtube.com/embed/DqUQW3xyQ1c?autoplay=1"
      title="video"
      width="560"
    />
  </div>
</div>
`;
